mitosheet/mitosheet/api/graph/plotly_express_graphs.py

```diff
@@ -0,0 +1,213 @@
+
+def graph_filtering(graph_type: str, df, column_headers):
+    """
+    Filteres the dataframe about to be graphed so that we don't crash the browser
+    """
+    return filter_df_to_safe_size(graph_type, df, column_headers)
+
+
+def graph_filtering_code(graph_type: str, df_name, df, column_headers) -> str:
+    """
+    Returns the code for filtering the dataframe so we don't crash the browser
+    """
+    # Check if we filter the graph
+    _, filtered = filter_df_to_safe_size(graph_type, df, column_headers)
+    
+    # If we don't filter the graph, then return an empty string
+    if not filtered: 
+        return ''
+    
+    # If we do filter the graph, then return the code needed to filter the graph
+    filtered_code = """from mitosheet import filter_df_to_safe_size
+
+# Filter the dataframe so that it does not crash the browser
+{df_name}_filtered, _ = filter_df_to_safe_size('{graph_type}', {df_name}, {column_headers})
+""".format(
+    graph_type=graph_type,
+    df_name=df_name, 
+    column_headers=column_headers
+)
+    return filtered_code
+
+
+def graph_creation(
+    graph_type: str, 
+    df: pd.DataFrame, 
+    x_axis_column_headers: List[ColumnHeader],
+    y_axis_column_headers: List[ColumnHeader]
+) -> go.Figure: 
+    """
+    Creates and returns the Plotly express graph figure
+    """
+    num_x_axis_column_headers = len(x_axis_column_headers)
+    num_y_axis_column_headers = len(y_axis_column_headers)
+    
+    # Create the parameters that we use to construct the graph
+    x_arg = None
+    if num_x_axis_column_headers == 1:
+        # Note: In the new interface, x will always have a length of 0 or 1
+        x_arg = x_axis_column_headers[0]
+    if num_x_axis_column_headers > 1:
+        x_arg = x_axis_column_headers
+
+    y_arg = None
+    if num_y_axis_column_headers == 1:
+        y_arg = y_axis_column_headers[0]
+    if num_y_axis_column_headers > 1:
+        y_arg = y_axis_column_headers
+    
+    if graph_type == BOX:
+        return px.box(df, x=x_arg, y=y_arg)
+    if graph_type == HISTOGRAM:
+        return px.histogram(df, x=x_arg, y=y_arg)
+    if graph_type == BAR:
+        return px.bar(df, x=x_arg, y=y_arg)
+    if graph_type == SCATTER:
+        return px.scatter(df, x=x_arg, y=y_arg)
+
+    
+def graph_creation_code(
+    graph_type: str, 
+    df_name: str, 
+    x_axis_column_headers: List[ColumnHeader],
+    y_axis_column_headers: List[ColumnHeader]
+) -> str:
+    """
+    Returns the code for creating the Plotly express graph
+    """
+    num_x_axis_column_headers = len(x_axis_column_headers)
+    num_y_axis_column_headers = len(y_axis_column_headers)
+
+    # Create the chords that we use to construct the graph
+    x_chord = ''
+    if num_x_axis_column_headers == 1:
+        # Note: In the new interface, x will always have a length of 0 or 1
+        x_chord = f'x="{x_axis_column_headers[0]}"'
```

Comment:
@aarondr77 there is a function called `transpile_column_header` or something similar that handles different types of column headers (e.g. not just strings) properly. There's a utility in the same file that handles lists as well, which could be nice below too (although I'm not sure that one is needed).